src/js/labels/queries.ts

```diff
@@ -1,12 +1,50 @@
-import { useQuery } from "@tanstack/react-query";
-import { fetchLabels } from "./api";
+import { ErrorResponse } from "@/types/types";
+import { Label } from "@labels/types";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { createLabel, fetchLabels, removeLabel, updateLabel } from "./api";
 
 export const labelQueryKeys = {
     all: () => ["label"] as const,
     lists: () => ["label", "list"] as const,
     list: (filters: Array<string | number | boolean>) => ["label", "list", "single", ...filters] as const,
 };
 
+/**
+ * Fetch a list of labels from the API
+ *
+ * @returns A list of labels
+ */
 export function useFetchLabels() {
-    return useQuery(labelQueryKeys.list([]), fetchLabels);
+    return useQuery<Label[]>(labelQueryKeys.list([]), fetchLabels);
+}
+
+/**
+ * Initializes a mutator for creating a label
```

Comment:
Phrasing nit:

```suggestion
 * Initialize a mutator for creating a label
```
apply to other mutators as well